google/cloud/storage/examples/storage_object_retention_samples.cc

```diff
@@ -0,0 +1,244 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/client.h"
+#include "google/cloud/storage/examples/storage_examples_common.h"
+#include "google/cloud/internal/getenv.h"
+#include <iostream>
+
+namespace {
+
+void InsertObjectWithRetention(google::cloud::storage::Client client,
```

Comment:
Do we not region tags for these samples yet?

Reply:
Candidly, I did not look.  Now that I did, I only found `storage_set_object_retention_policy` and `storage_create_bucket_with_object_retention`.